docs/us.md

```diff
@@ -0,0 +1,225 @@
+
+* **Confirmed**  
+If a proposal has been approved in some block and there are at least `k` blocks after this one, then
+update proposal is called _confirmed_.
+_Confirmed_ state reflects the fact that a proposal's state cannot be changed anymore 
+because we have guarantee that at most `k` blocks may be rolled back.
+
+### Software and block versions. Data associated with block version
+
+When a developer wants to fix some bug or add some features to application,
+which don't affect protocol rules, he should propose an update with new `SoftwareVersion`.
+```
+data SoftwareVersion = SoftwareVersion
+    { svAppName :: !ApplicationName
+    , svNumber  :: !NumSoftwareVersion
+    }
+```
+
+`SoftwareVersion` contains application name and numeric representation of application version.
+
+If a developer wants to change some protocol rules, for example, `Address` format, 
+he must propose update with new `BlockVersion`
+
+```
+data BlockVersion = BlockVersion
+    { bvMajor :: !Word16
+    , bvMinor :: !Word16
+    , bvAlt   :: !Word8
+    }
+```
+
+So block version essentially is a tuple `(Maj, Min, Alt)`.
+Block version is also called **protocol version**.
+
+There are some protocol constants which depend on `BlockVersion`, for example max block size, slot duration, etc.
+Update system was designed in such way that you must attach new values of this constants if they are updated.
+`BlockVersionModifier` reflects these changes:
+```
+data BlockVersionModifier = BlockVersionModifier
+    { bvmScriptVersion     :: !(Maybe ScriptVersion)
+    , bvmSlotDuration      :: !(Maybe Millisecond)
+    , bvmMaxBlockSize      :: !(Maybe Byte)
+    , bvmMaxHeaderSize     :: !(Maybe Byte)
+    , bvmMaxTxSize         :: !(Maybe Byte)
+    , bvmMaxProposalSize   :: !(Maybe Byte)
+    , bvmMpcThd            :: !(Maybe CoinPortion)
+    , bvmHeavyDelThd       :: !(Maybe CoinPortion)
+    , bvmUpdateVoteThd     :: !(Maybe CoinPortion)
+    , bvmUpdateProposalThd :: !(Maybe CoinPortion)
+    , bvmUpdateImplicit    :: !(Maybe FlatSlotId)
+    , bvmSoftforkRule      :: !(Maybe SoftforkRule)
+    , bvmTxFeePolicy       :: !(Maybe TxFeePolicy)
+    , bvmUnlockStakeEpoch  :: !(Maybe EpochIndex)
+    }
+```
+So if a value is going to be updated then a field is `Just` and `Nothing` otherwise.
+
+### Adoption of block version
+
+Assume a proposal which is bumping `BlockVersion` became _confirmed_.
+Though the proposal is already _confirmed_ along with block version but stakeholders
+haven't updated its software yet, hence, they can't validate block of _confirmed_ block version.
+
+To avoid a situation when adversary stakeholder issues invalid block of just _confirmed_ block version,
+others stakeholders will validate blocks according to previous **adopted** block version, 
+until _confirmed_ becomes **adopted**.
+
+Informally `BlockVersion` is **adopted** if sum of block issuers' stakes, 
+which issued blocks of this `BlockVersion` at least once, is a majority.
```

Comment:
Why majority? Not just majority, it depends on `SoftforkRule`, initially we want it to be 90% and after a month we want 60% (and never less than 60%).

Reply:
Well, maybe that's ok, because it's _informally_ and below you describe it more formally… 🤷‍♂️

Reply:
mostly copied from 
https://cardanodocs.com/cardano/update-mechanism/